common/lib/xmodule/xmodule/modulestore/tests/test_mixed_modulestore.py

```diff
@@ -436,6 +436,46 @@ def test_get_items(self, default_ms, max_find, max_send):
                 revision=ModuleStoreEnum.RevisionOption.draft_preferred
             )
 
+    @ddt.data(ModuleStoreEnum.Type.split, ModuleStoreEnum.Type.mongo)
+    def test_get_items_in_tree_only(self, default_ms):
+        """
+        Test `in_tree_only` option helps in returning only those items which are present in course tree.
+        """
+        self.initdb(default_ms)
+
+        test_course = self.store.create_course('testx', 'GreekHero', 'test_run', self.user_id)
+        course_key = test_course.id
+
+        # Check items found in start is 2
+        items = self.store.get_items(course_key)
+        self.assertEqual(len(items), 2)
+
+        # Check that orphans are not found
+        orphans = self.store.get_orphans(course_key)
+        self.assertEqual(len(orphans), 0)
+
+        # Add an orphan to test course
+        orphan = course_key.make_usage_key('chapter', 'OrphanChapter')
+        self.store.create_item(self.user_id, orphan.course_key, orphan.block_type, block_id=orphan.block_id)
+
+        # Check that now an orphan is found
+        orphans = self.store.get_orphans(course_key)
+        self.assertEqual(len(orphans), 1)
+
+        # Check now `get_items` retrieves an extra item added above which is an orphan.
+        items = self.store.get_items(course_key)
+        self.assertEqual(len(items), 3)
+
+        # Check now `get_items` with `in_tree_only` kwarg does not retrieves an orphan block.
+        items_in_tree_only = self.store.get_items(course_key, in_tree_only=True)
+        expected_found_items_in_tree_only = 2
```

Comment:
you can also test to see if the orphan in returned in `get_items`. You could set a ddt variable `orphan_in_items` that's set to True or False, then do something like:
`self.assertEqual(orphan in items_in_tree, oprhan_in_items)`